src/server.rs

```diff
@@ -171,37 +178,49 @@ struct Opts {
     /// location to store backup files
     #[clap(short, long)]
     dir: Option<String>,
-    /// alternative port location
+    /// alternative uds location
     #[clap(short, long)]
-    port: Option<std::net::SocketAddr>,
+    uds: Option<String>,
 }
 
 #[tokio::main]
 #[allow(unused)]
 pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let default_grpc_port = "[::1]:10000".parse().unwrap();
+    let default_uds_path = "/var/tmp/nv-dhcp".to_string();
     let default_config_dir = "".to_string();
 
     env_logger::builder().format_timestamp(None).init();
     let opts = Opts::parse();
 
     // where we store the cache file
-    let conf_dir = opts.dir.unwrap_or_else(|| default_config_dir);
+    let conf_dir = opts.dir.unwrap_or(default_config_dir);
     // location of the grpc port
-    let addr = opts.port.unwrap_or_else(|| default_grpc_port);
+    let uds_path = opts.uds.unwrap_or(default_uds_path);
+
+    tokio::fs::create_dir_all(Path::new(&uds_path).parent().unwrap()).await?;
```

Comment:
are both the unwrap and await protected here?  The ? applies to the await yes?

Reply:
The `?` applies to the `tokio::fs::create_dir_all`. If it returns an error it will return an Err(io::Result) from main. Would you like this error to be more verbose, instead of just returning the error?